databricks/koalas/tests/test_dataframe.py

```diff
@@ -1848,10 +1848,16 @@ def test_isin(self):
         kdf = ks.from_pandas(pdf)
 
         self.assert_eq(kdf.isin([4, "six"]), pdf.isin([4, "six"]))
+        # Seems like pandas has a bug when passing `np.array` as parameter
```

Comment:
`pandas` should be `koalas` :)

Reply:
Oh, seems like pandas also has bug when passing `np.array` to pandas `DataFrame.isna`.

For example, let's say we have pandas DataFrame as below.

```python
>>> pdf
          a  b      c
0.475873  4  1    one
0.659894  2  2  three
0.581484  3  9    six
0.226334  4  4  seven
0.349045  8  2    one
0.464269  6  4      5
```

Then use `isin` to check if each column of DataFrame has `4` or `"six"`.

```python
>>> pdf.isin(np.array([4, "six"]))
              a      b      c
0.475873  False  False  False
0.659894  False  False  False
0.581484  False  False   True
0.226334  False  False  False
0.349045  False  False  False
0.464269  False  False  False
```

I think this should be

```python
>>> pdf.isin([4, "six"])
              a      b      c
0.475873   True  False  False
0.659894  False  False  False
0.581484  False  False   True
0.226334   True   True  False
0.349045  False  False  False
0.464269  False   True  False
```

Reply:
Ohh... thanks, good catch!

This is because Koalas internally uses PySpark's `isin` directly,

but seems like PySpark's `isin` doesn't distinguish the type between `4` and `"4"`...

```python
>>> sdf
DataFrame[__index_level_0__: double, a: bigint, b: bigint, c: string, __natural_order__: bigint]

>>> sdf.select(sdf.a).show()
+---+
|  a|
+---+
|  4|
|  2|
|  3|
|  4|
|  8|
|  6|
+---+

>>> sdf.select(sdf.a.isin(4)).show()
+----------+
|(a IN (4))|
+----------+
|      true|
|     false|
|     false|
|      true|
|     false|
|     false|
+----------+

>>> sdf.select(sdf.a.isin('4')).show()  # I think this should return `false` for all values since the type of column `a` is `bigint`
+----------+
|(a IN (4))|
+----------+
|      true|
|     false|
|     false|
|      true|
|     false|
|     false|
+----------+
```

Reply:
It must be caused by type-coercion rule. We can leave it as-is. cc @HyukjinKwon

Reply:
Thanks, @ueshin :)